tests/run-make/reset-codegen-1/rmake.rs

```diff
@@ -0,0 +1,36 @@
+// When rustc receives 4 codegen-units, an output path and an emit flag all simultaneously,
+// this could cause an annoying recompilation issue, uselessly lengthening the build process.
+// This test tries to reproduce this issue with varied emit flags - if it passes, the codegen-units
+// are being appropriately reset to 1 after each pass.
```

Comment:
AFAICT, this isn't actually being checked. We're just checking here that `foo.rs` "compiles with various cli flags", am I understanding that correctly?

Reply:
As in, the original test also isn't checking what this comment is saying?

Reply:
You're right, I wrongfully assumed. The test is checking that the fix did not break compilation, not that the fix itself is doing what it is supposed to do (preventing the annoying re-compilation).